### ISSUE TITLE
KIWI - OBS Image build for Fedora 25 / 26 not booting correctly

### ISSUE THREAD

**Issue by mmohring** [SUSE/kiwi#443]
## Problem description

If you create a VMX image in OBS, the resulting Image gets stuck while booting

## Actual behaviour

I suspect it got never tested ?

The console of the VMX image shows it boots and then I suspect it ends in some loop in the initrd ? It shows the console output:

... searching for boot device in Application ID ...

## Steps to reproduce the behaviour

1. Add KIWI NG in the prjconf of your Image Project in KIWI.
2. Create an Image package for type VMX
3. Boot the Image

You can check this in API b.o.o @ Project home:MartinMohring:KIWI-NG:images

**Comment by mmohring** [comment]
Can you give me some advice what information you need and how can I grab it from a VMX image which seems to loop in initrd ? How can I debug this ?

**Comment by mmohring** [comment]
I have created two projects in b.o.o to help you understand this issue:

1. home:MartinMohring:KIWI-NG:Appliance
2. home:MartinMohring:KIWI-NG:images

1. contains python-kiwi + secondary packages needed by KIWI NG for Fedora / CentOS. 2. contains the test Images for Fedora which do not work, and the CentOS Image which works.

**Comment by schaefi** [comment]
Hmm

```xml
<type image="vmx" filesystem="ext4" boot="isoboot/fedora-25.0" ...
```

That is odd, you want to build a virtual disk image by using kiwi's live iso boot code ?
This is not going to work because the image type and the referenced boot image description does
not fit together.

We provide a working Fedora25 image description here:

* https://github.com/SUSE/kiwi-descriptions/tree/master/fedora/x86_64/fedora-25.0-JeOS

I built this one locally with:

```
$ sudo kiwi-ng --type vmx system build --description ~ms/Project/kiwi-descriptions/fedora/x86_64/fedora-25.0-JeOS/ --target-dir /tmp/mytest/
```

and it boots with:

```
$ sudo qemu-kvm LimeJeOS-Fedora-25.0.x86_64-1.2.0.raw
```

**Comment by mmohring** [comment]
@schaefi, how can I provide you with a patch for KIWI NG official working boot="vmxboot/fedora-25.0" config tree ? Its currently in a tar file boot_arch.tgz .

**Comment by mmohring** [comment]
@schaefi you closed this issue, but I was referring to use KIWI NG with OBS. Is KIWI NG not ready enough yet for using it with OBS, or is Fedora not officially supported by using KIWI NG in OBS ?

**Comment by mmohring** [comment]
@schaefi, I could track down this problem now. I used your templates instead of the initial .kiwi file. I use now dracut type of initrd. I tested with Fedora 25 / 26. I found:

* that I could create a booting and correctly starting up config.sh / config.kiwi
* I could get networking with DHCP start up in a apropriately setup KVM VM
* I found that dracut on Fedora 25 has a bug which prohibits complete startup of a VM
* Using the Fedora 25 Update repo fixed it
* I also rebuild a fixed dracut from Fedora 25 and put it to a Image not using Fedora 25 Updates, worked

I also checked that its not the kernel causing this, using the Fedora 25 release and updates kernels. Both did not work with old dracut, and both worked with a fixed dracut.

I also found out that after an update both on Fedora 25 and 26 with dnf update, the grub.cnf seems borked. I could fix it by hand with grub2-mkconfig > /boot/grub2/grub.cfg.

I am not sure what is the difference between calling KIWI NG via OBS or via command line like you do. Are you sure your images do really work ? I see them boot but stuck in a startup job detecting the disk. This does not happen with the fixed dracut or with the dracut from Fedora 25 Updates. It does also not happen with Fedora 26 at all, release or updates.
